Emby.Server.Implementations/Plugins/PluginManager.cs

```diff
@@ -105,18 +108,52 @@ public IEnumerable<Assembly> LoadAssemblies()
                     continue;
                 }
 
-                foreach (var file in plugin.DllFiles)
+                var pluginCount = 0;
+                int j = 0;
+                while (j < plugin.DllFiles.Count)
                 {
+                    var file = plugin.DllFiles[j];
                     Assembly assembly;
                     try
                     {
                         assembly = Assembly.LoadFrom(file);
 
                         // This force loads all reference dll's that the plugin uses in the try..catch block.
                         // Removing this will cause JF to bomb out if referenced dll's cause issues.
-                        assembly.GetExportedTypes();
+                        var ass = assembly.GetExportedTypes();
+
+                        // Backwards compatibility fix where we multiple assemblies containing types.
+                        var containsPlugin = ass.Any(a => pluginType.IsAssignableFrom(a));
+
+                        if (containsPlugin && ++pluginCount > 1)
+                        {
+                            _logger.LogWarning("Depreciated: Multiple plugins found in one folder : {Plugin}", file);
+                            plugin.DllFiles.RemoveAt(j);
+                            plugin.Manifest.Depreciated = true;
+                            // Clone LocalPlugin so that it can be managed.
+                            var p = new LocalPlugin(
+                                plugin.Path,
+                                true,
+                                new PluginManifest()
+                                {
+                                    Status = PluginStatus.Active,
+                                    Name = Path.GetFileNameWithoutExtension(file) ?? file,
+                                    AutoUpdate = false,
+                                    Id = file.GetMD5(),
```

Comment:
The file likely already has an ID, so we should probably use it?

Reply:
??
The original plugin has an ID which needs to be unique.
_file_ contains the string filename of the assembly that has been loaded. LocalPlugin.ID points to Manifest.ID, so it does appear anywhere else.